src/coreComponents/finiteVolume/FluxApproximationBase.cpp

```diff
@@ -71,19 +71,26 @@ void FluxApproximationBase::initializePreSubGroups()
       if( !(mesh.isShallowCopy() ) )
       {
         // Group structure: mesh1/finiteVolumeStencils/myTPFA
-
-        Group & stencilParentGroup = mesh.registerGroup( groupKeyStruct::stencilMeshGroupString() );
-        Group & stencilGroup = stencilParentGroup.registerGroup( getName() );
+        Group * stencilParentGroup = mesh.getGroupPointer( groupKeyStruct::stencilMeshGroupString() );
+        if ( stencilParentGroup == nullptr )
+        {
+          stencilParentGroup = &(mesh.registerGroup( groupKeyStruct::stencilMeshGroupString() ));
+        }
+        
+        Group & stencilGroup = stencilParentGroup->registerGroup( getName() );
```

Comment:
If I understand correctly based on what we discussed, this is needed because the simulation will have two stencils objects (stabilization and matrix), is that right? Maybe add a comment here to remind us that this situation can happen, because I had completely forgotten that it was possible.